lib/ansible/modules/cloud/vmware/vmware_guest_powerstate.py

```diff
@@ -110,7 +135,50 @@ def main():
 
     if vm:
         # VM already exists, so set power state
-        result = set_vm_power_state(pyv.content, vm, module.params['state'], module.params['force'])
+        scheduled_at = module.params.get('scheduled_at', None)
+        if scheduled_at:
+            powerstate = {
+                'powered-off': vim.VirtualMachine.PowerOff,
+                'powered-on': vim.VirtualMachine.PowerOn,
+                'reboot-guest': vim.VirtualMachine.RebootGuest,
+                'restarted': vim.VirtualMachine.Reset,
+                'shutdown-guest': vim.VirtualMachine.ShutdownGuest,
+                'suspended': vim.VirtualMachine.Suspend,
+            }
+            dt = ''
+            try:
+                dt = datetime.strptime(scheduled_at, '%d/%m/%Y %H:%M')
+            except ValueError as e:
+                module.fail_json(msg="Failed to convert given date and time string to Python datetime object,"
+                                     "please specify string in 'dd/mm/yyyy hh:mm' format: %s" % to_native(e))
+            schedule_task_spec = vim.scheduler.ScheduledTaskSpec()
+            schedule_task_spec.name = 'Schedule task for vm %s' % vm.name
+            schedule_task_spec.description = 'Schedule task for vm %s' % vm.name
+            schedule_task_spec.scheduler = vim.scheduler.OnceTaskScheduler()
+            schedule_task_spec.scheduler.runAt = dt
+            schedule_task_spec.action = vim.action.MethodAction()
+            schedule_task_spec.action.name = powerstate[module.params.get('state')]
+            schedule_task_spec.enabled = True
+
+            try:
+                pyv.content.scheduledTaskManager.CreateScheduledTask(vm, schedule_task_spec)
+                # As this is async task, we create scheduled task and mark state to changed.
+                module.exit_json(changed=True)
+            except vim.fault.InvalidName as e:
+                module.fail_json(msg="Failed to create scheduled task %s for %s : %s" % (module.params.get('state'),
+                                                                                         vm.name,
+                                                                                         to_native(e.msg)))
+            except vim.fault.DuplicateName as e:
+                module.fail_json(msg="Failed to create scheduled task %s for %s as task with "
```

Comment:
maybe you should include the `state` and the time (`dt`) in `schedule_task_spec.name`.  Then in case of DuplicateName you could run `module.exit_json(changed=False)`.
What do you think ?